lib/traject/dor_config.rb

```diff
@@ -0,0 +1,394 @@
+
+def parse_incipit(sdb)
+  sdb.smods_rec.related_item.each do |item|
+    item.note.each do |note|
+      return note.text.strip if note.attr('type') == 'incipit'
+    end
+  end
+  nil
+end
+
+def iiif_manifest_url(bare_druid)
+  format ::Settings.purl.iiif_manifest_url, druid: bare_druid
+end
+
+# @return [Array{String}] The IDs from geonames //subject/geographic URIs, if any
+def extract_geonames_ids(sdb)
+  sdb.smods_rec.subject.map do |z|
+    next unless z.geographic.any?
+    uri = z.geographic.attr('valueURI')
+    next if uri.nil?
+
+    m = %r{^https?://sws\.geonames\.org/(\d+)}i.match(uri.value)
+    m ? m[1] : nil
+  end.compact.reject(&:empty?)
+end
+
+# rubocop:disable Metrics/AbcSize
+# Fetch remote geonames metadata and format it for Solr
+# @param [String] id geonames identifier
+# @return [String] Solr WKT/CQL ENVELOPE based on //geoname/bbox
+def get_geonames_api_envelope(id)
+  url = "http://api.geonames.org/get?geonameId=#{id}&username=#{::Settings.geonames_username}"
+  xml = Nokogiri::XML Faraday.get(url).body
+  bbox = xml.at_xpath('//geoname/bbox')
+  return if bbox.nil?
+  min_x, max_x = [bbox.at_xpath('west').text.to_f, bbox.at_xpath('east').text.to_f].minmax
+  min_y, max_y = [bbox.at_xpath('north').text.to_f, bbox.at_xpath('south').text.to_f].minmax
+  "ENVELOPE(#{min_x},#{max_x},#{max_y},#{min_y})"
+rescue Faraday::Error => e
+  logger.error("Error fetching/parsing #{url} -- #{e.message}")
+  nil
+end
+# rubocop:enable Metrics/AbcSize
+
+# go grab the supplied file url, grab the file, encode and return
+# TODO: this should also be able to deal with .rtf and .xml files, scrubbing/converting as necessary to get plain text
+def get_file_content(file_url)
+  response = Faraday.get(file_url)
+  response.body.scrub.encode('UTF-8', invalid: :replace, undef: :replace, replace: '?').gsub(/\s+/, ' ')
+rescue
+  logger.error("Error indexing full text - couldn't load file #{file_url}")
+  nil
+end
+
+# these are the file locations where full txt files can be found at the object level
+# this method returns an array of fully qualified public URLs that can be accessed to find full text countent
+def object_level_full_text_urls(sdb)
+  files = []
+  object_level_full_text_filenames(sdb).each do |xpath_location|
+    files += sdb.public_xml.xpath(xpath_location).map do |txt_file|
+      "#{::Settings.stacks.file_url}/#{sdb.bare_druid}/#{txt_file['id']}"
+    end
+  end
+  files
+end
+
+# xpaths to locations in the contentMetadata where full text object level files can be found,
+#  add as many as you need, all will be searched
+def object_level_full_text_filenames(sdb)
+  [
+    # feigenbaum style - full text in .txt named for druid
+    "//contentMetadata/resource/file[@id=\"#{sdb.bare_druid}.txt\"]"
+  ]
+end
+
+# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+def file_ids(resource)
+  ids = []
+  if resource.content_metadata
+    if display_type(dor_content_type(resource)) == 'image'
```

Comment:
What would you think about assigning this to a temp variable, it looks like it's called again in the `elseif` below.